homeassistant/components/iaqualink/sensor.py

```diff
@@ -45,13 +49,21 @@ def native_unit_of_measurement(self) -> str | None:
     @property
     def native_value(self) -> int | float | None:
         """Return the state of the sensor."""
-        if self.dev.state == "":
+        if not isinstance(self.dev.state, str) or self.dev.state == "":
             return None
 
         try:
             return int(self.dev.state)
```

Comment:
Is it critical to check for `int` here? Could we get away with a single check for `float`?

Reply:
I can't remember the original intent but I suspect I didn't want decimals if they weren't needed.